build.zig

```diff
@@ -1,187 +1,93 @@
-    var steps = ArrayList(*Step).init(b.allocator);
-    steps.append(&grub_cmd.step) catch unreachable;
-    steps.append(&iso_cmd.step) catch unreachable;
-    return steps;
-}
-
-fn buildLink(b: *Builder, target: builtin.Arch, build_path: []const u8) *Step {
-    const exec = b.addExecutable("pluto.elf", null);
-    const elf_path = concat(b.allocator, build_path, "/iso/boot") catch unreachable;
-    exec.setOutputDir(elf_path.toSlice());
-    exec.setLinkerScriptPath("link.ld");
-    exec.setTarget(target, builtin.Os.freestanding, builtin.Abi.gnu);
-    for (src_files.toSlice()) |file| {
-        var file_obj = concat(b.allocator, build_path, "/") catch unreachable;
-        file_obj.append(file) catch unreachable;
-        file_obj.append(".o") catch unreachable;
-        exec.addObjectFile(file_obj.toSlice());
-        exec.setMainPkgPath(".");
-    }
-    for (src_files_asm.toSlice()) |file| {
-        var file_obj = concat(b.allocator, build_path, "/") catch unreachable;
-        file_obj.append(file) catch unreachable;
-        file_obj.append(".o") catch unreachable;
-        exec.addObjectFile(file_obj.toSlice());
+    const test_step = b.step("test", "Run tests");
+    if (rt_test) {
+        const script = b.addSystemCommand([_][]const u8{ "python3", "test/rt-test.py", "x86", b.zig_exe });
+        test_step.dependOn(&script.step);
+    } else {
+        const unit_tests = b.addTest(main_src);
+        unit_tests.setMainPkgPath(".");
+        unit_tests.addBuildOption(bool, "rt_test", rt_test);
+        test_step.dependOn(&unit_tests.step);
     }
-    return &exec.step;
-}
 
-fn buildObjects(b: *Builder, target: builtin.Arch, build_path: []const u8, src_path: []const u8) ArrayList(*std.build.LibExeObjStep) {
-    var objects = ArrayList(*std.build.LibExeObjStep).init(b.allocator);
-    const src_path2 = concat(b.allocator, src_path, "/") catch unreachable;
-    for (src_files.toSlice()) |file| {
-        var file_src = concat(b.allocator, src_path2.toSlice(), file) catch unreachable;
-        file_src.append(".zig") catch unreachable;
-        const obj = b.addObject(file, file_src.toSlice());
-        obj.setMainPkgPath(".");
-        obj.setOutputDir(build_path);
-        obj.setTarget(target, builtin.Os.freestanding, builtin.Abi.gnu);
-        objects.append(obj) catch unreachable;
-    }
-    for (src_files_asm.toSlice()) |file| {
-        var file_src = concat(b.allocator, src_path2.toSlice(), file) catch unreachable;
-        file_src.append(".s") catch unreachable;
-        const obj = b.addAssemble(file, file_src.toSlice());
-        obj.setOutputDir(build_path);
-        obj.setTarget(target, builtin.Os.freestanding, builtin.Abi.gnu);
-        objects.append(obj) catch unreachable;
-    }
-    return objects;
+    const debug_step = b.step("debug", "Debug with gdb");
+    const debug_cmd = b.addSystemCommand([_][]const u8{
+        "gdb",
+        "-ex",
+        "symbol-file",
+    });
+    debug_cmd.addArtifactArg(exec);
```

Comment:
Using `addArtifactArg` adds a dependency on building the kernel so it will be built every time you run debug. That ties in to #26 so it can be fixed later.